### ISSUE TITLE
ERROR: for azuracast_nginx_ Cannot start service nginx after latest update

### ISSUE THREAD

**Issue by suricloud** [AzuraCast/AzuraCast#1101]
**Installation method**
Docker

**Host OS (for Docker installations)**
Debian 9
**Describe the bug**

I do get this error when updating to the latest:
azuracast_mariadb_1 is up-to-date
azuracast_stations is up-to-date
azuracast_redis_1 is up-to-date
Creating azuracast_letsencrypt_1 ... done
Creating azuracast_web_1         ... done
Creating azuracast_cli_1         ...
Creating azuracast_nginx_1       ... error
Creating azuracast_cron          ...

ERROR: for azuracast_nginx_1  Cannot start service nginx: driver failed programm Creating azuracast_cli_1         ... done
33f89d71d8ea9df0fdfa94e1a9f60583a8a87eca9): Bind for 0.0.0.0:443 failed: port is Creating azuracast_cron          ... done

ERROR: for nginx  Cannot start service nginx: driver failed programming external  connectivity on endpoint azuracast_nginx_1 (d17a5fe3b9c0330dc6b071c33f89d71d8ea 9df0fdfa94e1a9f60583a8a87eca9): Bind for 0.0.0.0:443 failed: port is already all ocated
ERROR: Encountered errors while bringing up the project.
Update complete!

**To Reproduce**

**Expected behavior**

**Relevant Logs**

**Screenshots**

**Device(s):**
 - Device:  
 - OS: 
 - Browser: 
 - Version: 

**Additional context**

**Comment by Vaalyn** [comment]
There is some other process on your system that is already using the port 443. See this line:
```
Bind for 0.0.0.0:443 failed: port is already allocated
```

That will most likely be another web server like apache or nginx. You can check which process is listening on this port with the command `lsof -i`

**Comment by SlvrEagle23** [comment]
@suricloud You should make sure you update your `docker-compose.yml` file and any override files that may also reference services. We have rolled out a new consolidated image for the `web` service that replaces `web`, `nginx`, `cron`, `cli` and `letsencrypt` (as they were all very similar and consisted of mostly duplicated functionality). If you're still seeing a `nginx` service trying to spin up, then some override file is trying to spin up the previous nginx service right on top of the new web service, which actually handles the web traffic now.

**Comment by williambatson** [comment]
Boy am I in deep water -  ran the latest update and all i got was the welcome to ngix message at my url -   I freaked out of course.   Just gave up and re-installed Ubuntu 16.04 -  Docker install fails - 
New docker-compose.yml file loaded.
Error: No such volume: azuracast_www_data
Error: No such volume: azuracast_tmp_data
/usr/local/bin/docker-compose: line 1: Not: command not found

Traditional install works and completes but again -  URL only shows nginx welcome screen.

So in short - I toasted me server -    What am I doing wrong here?

Thanks!

**Comment by SlvrEagle23** [comment]
@williambatson You didn't hose the server, it's just a problem with not having git and curl installed before running the Docker installer.

I'm going to make some changes to make sure those are checked before running the installer, as we've had a number of instances where installations have gone awry just like that.

**Comment by SlvrEagle23** [comment]
@williambatson I've added some extra safety checks into the `docker.sh` installer script that should avoid problems like that in the future. You can run `./docker.sh update-self` and try to run the installer again and it should work fine.

**Comment by williambatson** [comment]
yep - that fixed the issue and was able to re-install and recover.
I know it's been talked about before -  but isn't there an active change log posted somewhere so we know whether or not subsequent updates are needed or advised?
But as always - thanks for the help and work, much appreciated!

**Comment by williambatson** [comment]
I posted my reply before refreshing the page to see your reply -  I've always preferred traditional over docker due to disk space - my thinking was if I upgrade or downgrade my VPS and the disk size changes, docker would not grow with the increased server size and really not work at all if I downsized.    Is my thinking correct on this?  Or is Docker able to expand to allow for more disk usage easily if I upgrade my server?  I know you recommend Docker, just curious why.    Thanks again!

**Comment by SlvrEagle23** [comment]
@williambatson Glad to hear that resolved the issue.

The main reason for supporting the Docker installation method is because we can very precisely configure the arrangement between all of our constituent software pieces, define that in a prebuilt image and then ensure that _everyone_ across thousands of installations has the exact same configuration. By comparison, Traditional installs have to coexist with any other software on your host device, which often causes unexpected problems and a higher maintenance burden. Given how few of us are maintaining the project and fielding GitHub issues, we're inclined to always recommend the option that is both more likely for the new user to work and less likely to result in getting stuck with an interoperability problem.

With regard to how Docker handles size adjustments, I wouldn't know as I haven't played with it in that capacity, but most of Docker's configuration can be edited through flatfiles on the host machine.
